File: Escola/Cursos.Api/scripts/service/TipoeventoService.js
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/13/2017 1:58:38 PM
///// Gerado automaticamente para alteracoes dos metodos comum alterar o template
///// </sumary>
"use strict";
app.factory("TipoeventoService", ["$http", "ngAuthSettings", function ($http, ngAuthSettings) {
        var service = {};
		var serviceBase = ngAuthSettings.apiServiceBaseUri;
        service.GetAll = function (callback) {
            $http.get(serviceBase + "Tipoevento/getall")
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.GetById = function (id, callback) {
            $http.get(serviceBase + "Tipoevento/getbyid/" + id)
               .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Post = function (Tipoevento, callback) {
            $http.post(serviceBase + "Tipoevento/post", Tipoevento)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Put = function (Tipoevento, callback) {
            $http.post(serviceBase + "Tipoevento/put", Tipoevento)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

		service.Delete = function (Tipoevento, callback) {
            $http.post(serviceBase + "Tipoevento/delete", Tipoevento)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        return service;
    }]);

File: Escola/Cursos.Repository/Services/TipoeventoService.cs
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 9:23:24 PM

///// Para outros metodos implementar classe partial
///// </sumary>
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;
using Cursos.Repository.Services.Base;
using Cursos.Data;
using Cursos.Repository.Services.Interfaces;
using System.Linq.Expressions;
using Cursos.Repository.Work.Interface;
namespace  Cursos.Repository.Services
{
	  /// <summary>
      /// Public Service Class  TipoeventoService
      /// </summary>
      public partial class TipoeventoService : BaseServices, ITipoeventoService
      {

		public TipoeventoService(IUnitOfWork unitOfWork)
        {
			UnitOfWork = unitOfWork;
        }

		/// <summary>
        /// Add Range Entity EV_TIPOEVENTO
        /// </summary>
        /// <param name="entity"></param>
		public void Add(IEnumerable<EV_TIPOEVENTO> entity)
        {
            UnitOfWork.TipoeventoRepository.Add(entity);
        }

		/// <summary>
        /// Add Entity EV_TIPOEVENTO
        /// </summary>
        /// <param name="entity"></param>
		public void Add(EV_TIPOEVENTO entity)
        {	
            UnitOfWork.TipoeventoRepository.Add(entity);
        }
		/// <summary>
        /// Delete Entity EV_TIPOEVENTO
        /// </summary>
        /// <param name="entity"></param>
        public void Delete(EV_TIPOEVENTO entity)
        {
            UnitOfWork.TipoeventoRepository.Delete(entity);
        }
		 /// <summary>
        /// Get All Entity EV_TIPOEVENTO
        /// </summary>
        /// <returns>IEnumerable EV_TIPOEVENTO</returns>
        public IEnumerable<EV_TIPOEVENTO> GetAll()
        {
			return UnitOfWork.TipoeventoRepository.GetAll();
        }
	     
		/// <summary>
        /// Update Entity EV_TIPOEVENTO
        /// </summary>
        /// <param name="entity"></param>
        public void Update(EV_TIPOEVENTO entity)
        {
            UnitOfWork.TipoeventoRepository.Edit(entity);
        }

		/// <summary>
        /// Commit Transaction
        /// </summary>
        /// <returns></returns>
        public int Commit()
        {
            return UnitOfWork.Commit();
        }
		/// <summary>
        /// Commit Async Transaction
        /// </summary>
		public Task<int> CommitAsync()
        {
            return UnitOfWork.CommitAsync();
        }
				/// <summary>
        /// Get Entity EV_TIPOEVENTO By Id
        /// </summary>
        /// <param name="id"></param>
        /// <returns>EV_TIPOEVENTO</returns>
		public EV_TIPOEVENTO GetById(int id)
        {
			return UnitOfWork.TipoeventoRepository.FindBy(x => x.ID.Equals(id)).SingleOrDefault();
        }		
				
		public IEnumerable<EV_TIPOEVENTO> FindBy(Expression<Func<EV_TIPOEVENTO, bool>> predicate)
        {
            return UnitOfWork.TipoeventoRepository.FindBy(predicate);
        }
      }
}
